### PR TITLE
Add support for UWS

### LINKED ISSUE #158: Support for the uws module
I would like to see `uws` support as it could allow many more clients to connect to multiogar while using less system resources.

### PR DESCRIPTION
**Pull request by henopied**
Fixes #158, adds a config option for uws/ws.

### CODE REVIEW

**Comment by SimonOrJ** [comment]
It'd be cleaner if you committed the code formatting cleanup and support for uws in separate commits for easier code reading/review.  That aside... hooray!

File: src/gameserver.ini [hunk 1]
```diff
@@ -27,6 +27,7 @@ logFileVerbosity = 5
 # serverMinionInterval: minion detection interval in milliseconds
 # serverPort: Server port which will be used to listen for incoming connections
 # serverBind: Server network interface which will be used to listen for incoming connections (0.0.0.0 for all IPv4 interfaces)
+# serverWsMode: Determines if the server will use UWS or WS, some features may not work with UWS, 0 = ws, 1 = uws
```
Comment (NatsuTheGreat) [inline src/gameserver.ini]: What kind of features?

Comment (SimonOrJ) [inline src/gameserver.ini]: I'm assuming Windows compatibility.

**Comment by NatsuTheGreat** [comment]
The format clean up wasn't even a clean up, it was some bug that made now change.

**Comment by SimonOrJ** [comment]
It's most likely the removal of the DOS line breaks (`^M`) and spaces.

File: src/gameserver.ini (hunk 1 above)
Comment (henopied) [inline src/gameserver.ini]: And in the current version of node apparently ssl too.

**Comment by henopied** [comment]
Sorry, my text editor decided to do that... I hope you wont mind the only thing i really changed was when the WebSocket client's prototype was modified and the addition of `loadWebSocketModule()`

**Comment by David1ali12** [comment]
it crashes after 10 mins of running but its much faster
```
[FATAL] TypeError: this.emit is not a function
    at Socket.WebSocket.sendPacket (/home/ubuntu/workspace/src/GameServer.js:1620:18)
    at PlayerTracker.sendUpdate (/home/ubuntu/workspace/src/PlayerTracker.js:395:17)
    at GameServer.updateClients (/home/ubuntu/workspace/src/GameServer.js:512:39)
    at GameServer.mainLoop (/home/ubuntu/workspace/src/GameServer.js:620:10)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by ItzLevvie** [comment]
^^ I've been running my servers for about a day and each server have no errors.

**Comment by ItzLevvie** [comment]
No errors on my end :)

**Comment by henopied** [comment]
Wait til a connection drops, you may have some issues then.

**Comment by ItzLevvie** [comment]
^^ I'm using custom libraries that's why I don't have any errors.

**Comment by henopied** [comment]
Fixed in latest uws!

**Comment by ItzLevvie** [comment]
You might want to update your PR

**Comment by henopied** [comment]
the pr asks for the latest uws

**Comment by ItzLevvie** [comment]
I'm talking about the PR conflicts.

**Comment by David1ali12** [comment]
@henopied its fixed now ? should I reinstall uws ?

**Comment by ItzLevvie** [comment]
Yes

**Comment by henopied** [comment]
Whoops I was having some git issues sorry I will remake this shortly.

**Pull request closed without merging**